Client/WebFilters/FilterListResourceDownloader.swift

```diff
@@ -0,0 +1,483 @@
+      isEnabled: filterList.isEnabled,
+      componentId: filterList.componentId
+    )
+    
+    if filterList.isEnabled {
+      register(filterList: filterList)
+    } else {
+      unregister(filterList: filterList)
+    }
+  }
+  
+  /// Register all enabled filter lists
+  private func registerAllEnabledFilterLists() {
+    for filterList in filterLists {
+      guard filterList.isEnabled else { continue }
+      register(filterList: filterList)
+    }
+  }
+  
+  /// Register this filter list and start all additional resource downloads
+  private func register(filterList: FilterList) {
+    guard adBlockServiceTasks[filterList.uuid] == nil else { return }
+    guard let adBlockService = adBlockService else { return }
+    guard let index = filterLists.firstIndex(where: { $0.uuid == filterList.uuid }) else { return }
+    
+    for resource in filterList.resources {
+      startFetching(resource: resource, for: filterList, index: index)
+    }
+
+    adBlockServiceTasks[filterList.uuid] = Task.detached(priority: .background) {
+      for await folderURL in await adBlockService.register(filterListUUID: filterList.uuid) {
+        guard let folderURL = folderURL else { continue }
+        guard await self.isEnabled(filterListUUID: filterList.uuid) else { return }
+        await self.handle(downloadedFolderURL: folderURL, forFilterListUUID: filterList.uuid, index: index)
+        
+        // Save the downloaded folder for later (caching) purposes
+        Task { @MainActor in
+          self.settingsManager.set(folderURL: folderURL, forUUID: filterList.uuid)
+        }
+      }
+    }
+  }
+  
+  /// Unregister, cancel all of its downloads and remove any `ContentBlockerManager` and `AdBlockEngineManager` resources for this filter list
+  private func unregister(filterList: FilterList) {
+    adBlockServiceTasks[filterList.uuid]?.cancel()
+    adBlockServiceTasks.removeValue(forKey: filterList.uuid)
+    
+    for resource in filterList.resources {
+      stopFetching(resource: resource)
+    }
+    
+    Task {
+      await withTaskGroup(of: Void.self) { group in
+        group.addTask {
+          await ContentBlockerManager.shared.removeResource(for: .filterList(uuid: filterList.uuid))
+        }
+        group.addTask {
+          await AdBlockEngineManager.shared.removeResources(for: .filterList(uuid: filterList.uuid))
+        }
+      }
+    }
+  }
+  
+  /// Start fetching the resource for the given filter list
+  private func startFetching(resource: ResourceDownloader.Resource, for filterList: FilterList, index: Int) {
+    guard fetchTasks[resource] == nil else {
+      // We're already fetching for this filter list
+      return
+    }
+    
+    fetchTasks[resource] = Task.detached(priority: .background) {
+      if let fileURL = ResourceDownloader.downloadedFileURL(for: resource) {
+        await self.handle(downloadedFileURL: fileURL, for: resource, filterListUUID: filterList.uuid, index: index)
+      }
```

Comment:
Change to non-detached task unless you need `ResourceDownloader.downloadedFileURL` to execute off main. Remove background priority unless you're fine with this possibly never executing